src/io/ipc/read/common.rs

```diff
@@ -286,6 +292,61 @@ pub fn read_dictionary<R: Read + Seek>(
     Ok(())
 }
 
+/// A small wrapper around `[Vec<u8>]` that allows us to reuse memory once it is initialized.
+/// This may improve performance of the `[Read]` trait.
+#[derive(Clone, Default)]
+pub struct ReadBuffer {
+    data: Vec<u8>,
+    // length to be read or is read
+    length: usize,
+}
+
+impl ReadBuffer {
+    /// Create a new [`ReadBuf`] initialized to `length`
+    pub fn new(length: usize) -> Self {
+        let data = vec![0; length];
+        Self { data, length }
+    }
+
```

Comment:
```suggestion
```

imo there is no need for this method - `From<Vec` covers it 👍

Reply:
Ok, will remove that.